File: app/src/main/java/com/example/wavesoffood/LoginActivity.java
package com.example.wavesoffood;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class LoginActivity extends AppCompatActivity {
    EditText email,pass;
    TextView tv_newuser;
    Button btn_google,btn_facebook,btn_login;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_login);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        email = findViewById(R.id.et_email);
        pass = findViewById(R.id.et_pass);
        btn_facebook = findViewById(R.id.btn_facebook);
        btn_google = findViewById(R.id.btn_google);
        btn_login = findViewById(R.id.btn_Login);
        tv_newuser = findViewById(R.id.tv_newuser);

        btn_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (email.getText().toString().trim().isEmpty()){

                    email.setBackground(ContextCompat.getDrawable(LoginActivity.this, R.drawable.error_et_bg));

                    return ;
                }
                if (pass.getText().toString().trim().isEmpty()){

                    pass.setBackground(ContextCompat.getDrawable(LoginActivity.this, R.drawable.error_et_bg));

                    return ;
                }

                Intent i = new Intent(LoginActivity.this, MainActivity.class);
                startActivity(i);

            }
        });
        tv_newuser.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(LoginActivity.this, SignupActivity.class);
                startActivity(i);
            }
        });


    }
}

File: app/src/main/java/com/example/wavesoffood/ui/user/LoginActivity.kt
package com.example.wavesoffood.ui.user

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.wavesoffood.R

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_login)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val et_email = findViewById<EditText>(R.id.et_email)
        val et_pass = findViewById<EditText>(R.id.et_pass)
        val tv_forgetPass = findViewById<TextView>(R.id.tv_forgetPass)
        val tv_singup = findViewById<TextView>(R.id.tv_singup)
        val btn_login = findViewById<Button>(R.id.btn_login)


        btn_login.setOnClickListener{
            val email = et_email.text.toString().trim()
            val pass = et_pass.text.toString().trim()
            if (email.isEmpty()){
                et_email.error = "require"
                return@setOnClickListener
            }
            if (pass.isEmpty()){
                et_pass.error = "required"
                return@setOnClickListener
            }
            val intent = Intent(applicationContext,MainActivity::class.java)
            startActivity(intent)
        }
        tv_singup.setOnClickListener{
            val intent = Intent(applicationContext,SignupActivity::class.java)
            startActivity(intent)
        }
        tv_forgetPass.setOnClickListener{
            val intent = Intent(applicationContext,ForgetPasswordActivity::class.java)
            startActivity(intent)
        }




    }
}